main/main.c

```diff
@@ -1,35 +1,57 @@
-#include <stdio.h>
-#include "low_power_mode.h"
 #include "freertos/FreeRTOS.h"
 #include "freertos/task.h"
+#include "esp_err.h"
+#include "esp_log.h"
+
+#include "bme280.h"
+#include "low_power_mode.h"
 
-/**
- * @brief The task that enters into the low power mode.
- *
- * FreeRTOS task that enters into the low power mode. It sets the sleep time using the `low_power_mode_set_sleep_time` function and then enters into the low power mode using the `low_power_mode_enter_deep_sleep` function.
- */
-void low_power_mode_task()
+#define TAG_BME280 "BME280"
+#define TAG "MAIN"
+
+void main_task()
 {
-    low_power_mode_set_sleep_time(CONFIG_LOW_POWER_MODE_SLEEP_TIME_SEC);
-    for (;;)
+    uint32_t com_rslt;
+
+    com_rslt = bme280_init_driver(CONFIG_BME280_I2C_ADDRESS);
+    com_rslt += bme280_set_oversamp(3, 3, 3);
+    com_rslt += bme280_set_settings(0, 4, 2);
+
+    ESP_LOGI(TAG_BME280, "BME280 init result: %ld", com_rslt);
+    if (com_rslt == 0)
     {
-        vTaskDelay(5000 / portTICK_PERIOD_MS);
-        low_power_mode_enter_deep_sleep();
+        ESP_LOGI(TAG_BME280, "BME280 init success");
+        for (int i = 1;; i++)
+        {
+            ESP_LOGI(TAG_BME280, "Temperature: %.2f C, Pressure: %.2f hPa, Humidity: %.2f %%", bme280_read_temperature(), bme280_read_pressure() / 100, bme280_read_humidity());
+            vTaskDelay(1000 / portTICK_PERIOD_MS);
+            if (i % 10 == 0)
+            {
+                low_power_mode_set_sleep_time(10);
+                low_power_mode_enter_deep_sleep();
+            }
+        }
+    }
+    else
+    {
+        ESP_LOGI(TAG_BME280, "BME280 init failed. code: %ld", com_rslt);
+        esp_restart();
     }
 }
 
-/**
- * @file main.c
- *
- * @brief This file contains the main function.
- */
-
-/**
- * @brief This function creates a task that enters into the low power mode after 5 seconds.
- *
- * This function is the entry point of the program. It creates a task that enters into the low power mode after 5 seconds.
- */
+// Main application
 void app_main(void)
 {
-    xTaskCreate(low_power_mode_task, "deep_sleep_task", 2048, NULL, 5, NULL);
+    TaskHandle_t main_task_handle = NULL;
+    xTaskCreate(&main_task, "main_task", 4096, NULL, 4, &main_task_handle);
```

Comment:
Add error handling in case that task can't be created.

> If the task was created successfully then pdPASS is returned. Otherwise
> errCOULD_NOT_ALLOCATE_REQUIRED_MEMORY is returned.

See [xTaskCreate documentation](https://www.freertos.org/a00125.html)